client/admin.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Marc Berhault (marc@cockroachlabs.com)
+
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/cockroachdb/cockroach/base"
+	"github.com/cockroachdb/cockroach/util"
+	"github.com/cockroachdb/cockroach/util/log"
+)
+
+const (
+	// Types of admin config to manipulate.
+
+	// Accounting expects proto.AcctConfig
+	Accounting = "acct"
+	// Permission expects proto.PermConfig
+	Permission = "perms"
+	// Quit only handles Get requests.
+	Quit = "quit"
+	// User expects proto.UserConfig
+	User = "users"
+	// Zone expects proto.ZoneConfig.
+	Zone = "zones"
+)
+
+// AdminClient issues http requests to admin endpoints.
+// TODO(marc): unify the way we handle addresses in clients.
+type AdminClient struct {
+	context    *base.Context
+	address    string
+	configType string
+}
+
+// NewAdminClient returns a new 'configType' admin client, talking to 'address'.
+func NewAdminClient(ctx *base.Context, address, configType string) *AdminClient {
```

Comment:
I have a question that is unrelated to this change, but was an issue while reading this PR: can we rename `base.Context` to something more descriptive?  This is especially bad because we have committed to using google's [request Context](http://godoc.org/golang.org/x/net/context) object, which is always passed as the first argument to functions.  

Here the first argument is named `ctx` and is of type `Context`, but it's not a request context; this tripped me up a little.